### PR TITLE
Updates to README; permissioning feature added

### PR DESCRIPTION
**Pull request by lparis**
@mcqueary @ColinSullivan1 @kozlovic @aricart @derekcollison @wallyqs

### CODE REVIEW

**Comment by aricart** [comment]
Looks good to me.

**Comment by ColinSullivan1** [comment]
LGTM - you'll want to fix the typo in the PR title.

File: README.md [hunk 1]
```diff
@@ -24,9 +54,15 @@ The NATS server is lightweight and very performant. Starting one with no argumen
...
 ```
 
-The server will be started and listening for client connections on port 4222 (the default) from all available interfaces. The logs will be displayed to stdout
-as shown above. There are a large range of supported clients that can be found at [https://nats.io/clients](https://nats.io/download).
-The server uses a text based protocol, so interacting with it can be as simple as using telnet.
+The server is started and listening for client connections on port 4222 (the default) from all available interfaces. The logs are displayed to stdout as shown above in the server output.
+
+### Clients
+
+The NATS ecosystem provides a large range of supported and community [clients](http://nats.io/documentation/clients/nats-clients/), including Go, Java, Node, and many more. For the complete up-to-date list, visit the [NATS download site]](https://nats.io/download).
```
Comment (derekcollison) [inline README.md]: Extra ]

File: README.md [hunk 2]
```diff
@@ -3,19 +3,49 @@
 
 A High Performance [NATS](https://nats.io) Server written in [Go.](http://golang.org)
 
-**Note**: The `master` branch may be in an *unstable or even in a broken state* during development. Please use [releases][github-release] instead of the `master` branch in order to get stable binaries.
+## Quickstart
```
Comment (derekcollison) [inline README.md]: Feels like Quickstart would be to download a binary. This needs Go installed which is not quick unless you already have it. Might consider combining this with binary download. We could say "If you have Go installed and want to run directly from tip, ..."

File: README.md [hunk 3]
```diff
@@ -417,220 +592,31 @@ Add into the server configuration file's authorization section.
 
 ## Monitoring
 
-If the monitoring port is enabled, the server will run a lightweight http server that has several endpoints defined, **[/varz, /connz, /routez, /subsz]**. All endpoints return a JSON object.
```
Comment (wallyqs) [inline README.md]: Think we should leave some of the monitoring endpoint snippets sections (such as section on CORS) or cover with alternate docs IMO

File: README.md [hunk 4]
```diff
@@ -417,220 +592,31 @@ Add into the server configuration file's authorization section.
...
-      "in_bytes": 0,
-      "out_bytes": 0,
-      "subscriptions": 0
-    }
-  ]
-}
-```
-
-### /subsz
+If the monitoring port is enabled, the NATS server runs a lightweight HTTP server that has the following endpoints: /varz, /connz, /routez, and /subsz. All endpoints return a JSON object. See [NATS Server monitoring](http://nats.io/documentation/server/gnatsd-monitoring/) for endpoint examples.
```
Comment (wallyqs) [inline README.md]: There is `/stacksz` too in case should be added (does not return a JSON object though).

File: README.md (hunk 1 above)
Comment (lparis) [inline README.md]: done

File: README.md (hunk 4 above)
Comment (kozlovic) [inline README.md]: Do not document that for now. This is for debugging purposes and should not be advertised.

**Comment by kozlovic** [comment]
LGTM

File: README.md (hunk 2 above)
Comment (lparis) [inline README.md]: done

File: README.md (hunk 3 above)
Comment (lparis) [inline README.md]: This is in the Monitoring section of the docs in order to reduce the size of the readme.

File: README.md (hunk 4 above)
Comment (lparis) [inline README.md]: done

**Pull request merged**